ucsm_apis/admin/tacacsplus.py

```diff
@@ -0,0 +1,437 @@
+    if mo is None:
+        raise UcsOperationError(caller,
+                        "Tacacsplus  Provider Group '%s' does not exist" % dn)
+    return mo
+
+
+def tacacsplus_provider_group_exists(handle, name, **kwargs):
+    """
+    checks if a tacacsplus provider group exists
+
+    Args:
+        handle (UcsHandle)
+        name (string): name of tacacsplus provider group
+        **kwargs: key-value pair of managed object(MO) property and value, Use
+                  'print(ucscoreutils.get_meta_info(<classid>).config_props)'
+                  to get all configurable properties of class
+
+    Returns:
+        (True/False, MO/None)
+
+    Example:
+        tacacsplus_provider_group_exists(handle, name="test_prov_grp")
+    """
+    try:
+        mo = tacacsplus_provider_group_get(handle, name,
+                                    caller="tacacsplus_provider_group_exists")
+    except UcsOperationError:
+        return (False, None)
+    mo_exists = mo.check_prop_match(**kwargs)
+    return (mo_exists, mo if mo_exists else None)
+
+
+def tacacsplus_provider_group_delete(handle, name):
+    """
+    deletes a tacacsplus provider group
+
+    Args:
+        handle (UcsHandle)
+        name (string): name of tacacsplus provider group
+
+    Returns:
+        None
+
+    Raises:
+        UcsOperationError: If AaaProviderGroup is not present
+
+    Example:
+        tacacsplus_provider_group_delete(handle, name="test_prov_grp")
+    """
+    mo = tacacsplus_provider_group_get(handle, name,
+                                    caller="tacacsplus_provider_group_delete")
+    handle.remove_mo(mo)
+    handle.commit()
+
+
+def tacacsplus_provider_group_provider_add(handle, group_name, name,
+                                           order="lowest-available",
+                                           descr=None, **kwargs):
+    """
+    adds a tacacsplus provider to a tacacsplus provider  group
+
+    Args:
+        handle (UcsHandle)
+        group_name (string): name of tacacsplus provider group
+        order (string): order
+        name (string): name of tacacsplus provider
+        descr (string): descr
+        **kwargs: Any additional key-value pair of managed object(MO)'s
+                  property and value, which are not part of regular args.
+                  This should be used for future version compatibility.
+    Returns:
+        AaaProviderRef: Managed Object
+
+    Raises:
+        UcsOperationError: If AaaProviderGroup Or AaaProvider is not present
+
+    Example:
+        tacacsplus_provider_group_provider_add(handle,
+                                    group_name="test_prov_grp",
+                                    name="test_tacac_prov")
+    """
+
+    from ucsmsdk.mometa.aaa.AaaProviderRef import AaaProviderRef
+
+    provider_group = tacacsplus_provider_group_get(handle, group_name,
+                            caller="tacacsplus_provider_group_provider_add")
+    provider = tacacsplus_provider_get(handle, name,
```

Comment:
It is not to be used afterwards but it should exist. Reference is being created.